components/script/script_module.rs

```diff
@@ -78,7 +78,7 @@ use crate::task::TaskBox;
 
 fn gen_type_error(global: &GlobalScope, string: String) -> RethrowError {
     rooted!(in(*GlobalScope::get_cx()) let mut thrown = UndefinedValue());
-    Error::Type(string).to_jsval(GlobalScope::get_cx(), global, thrown.handle_mut());
+    Error::Type(string).to_jsval(GlobalScope::get_cx(), global, thrown.handle_mut(), CanGc::note());
```

Comment:
Let's add a CanGc argument to this method and pass it here instead of calling CanGc::note.